packages/react/src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,426 @@
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+import { Ref } from '@stardust-ui/react-component-ref'
+
+import {
+  UIComponentProps,
+  createShorthandFactory,
+  ShorthandFactory,
+  commonPropTypes,
+  applyAccessibilityKeyHandlers,
+  childrenExist,
+  ChildrenComponentProps,
+  getOrGenerateIdFromShorthand,
+  AutoControlledComponent,
+  debounce,
+} from '../../lib'
+import { WithAsProp, withSafeTypeForAs, ShorthandCollection, ShorthandValue } from '../../types'
+import Button, { ButtonProps } from '../Button/Button'
+import CarouselItem, { CarouselItemProps } from './CarouselItem'
+import Text from '../Text/Text'
+import CarouselNavigation, { CarouselNavigationProps } from './CarouselNavigation'
+import CarouselNavigationItem, { CarouselNavigationItemProps } from './CarouselNavigationItem'
+
+export interface CarouselSlotClassNames {
+  itemsContainer: string
+  paddleNext: string
+  paddlePrevious: string
+  pagination: string
+  navigation: string
+}
+
+export interface CarouselProps extends UIComponentProps, ChildrenComponentProps {
+  /** Index of the currently active item. */
+  activeIndex?: number | string
+
+  /**
+   * Sets the aria-roledescription attribute.
+   */
+  ariaRoleDescription?: string
+
+  /** Specifies if the process of switching slides is circular. */
+  circular?: boolean
+
+  /** Initial activeIndex value. */
+  defaultActiveIndex?: number | string
+
+  /**
+   * Message generator for item position in the carousel. Used to generate
+   * individual i11y messages for items, such as "1 of 4", and for the
+   * pagination footer, in case there are no navigation tabs.
+   */
+  getItemPositionText?: (index: number, size: number) => string
+
+  /** Shorthand array of props for CarouselItem. */
+  items?: ShorthandCollection<CarouselItemProps>
+
+  /** Shorthand array of props for the buttons of the tabs navigation. */
+  navigation?:
+    | ShorthandValue<CarouselNavigationProps>
+    | ShorthandCollection<CarouselNavigationItemProps>
+
+  /**
+   * A Carousel can position its navigation to the bottom by default,
+   * to the top, to the left or to the right of the content.
+   */
+  navigationPosition?: 'bottom' | 'top' | 'left' | 'right'
+
+  /** A Carousel's paddles may fade away when mouse is not hovering the code. */
+  paddlesFade?: boolean
+
+  /** Shorthand for the paddle that navigates to the next item. */
+  paddleNext?: ShorthandValue<ButtonProps>
+
+  /**
+   * A Carousel can position its paddels inside the content, outside or inline
+   * with the navigation component.
+   */
+  paddlesPosition?: 'inside' | 'outside' | 'inline'
+
+  /** Shorthand for the paddle that navigates to the previous item. */
+  paddlePrevious?: ShorthandValue<ButtonProps>
+
+  /**
+   * On slide transition, the Carousel may translate the slides' position,
+   * fade their appearance or just hide and show without any animation.
+   */
+  slideTransition?: 'translate' | 'fade' | 'display'
```

Comment:
Let's not add this at this moment, and implement in the styles whatever is necessary for the team. I don't think one application will have different slideTransition. If this turns out to be a case, we can add it later.